### PR TITLE
Reader: Stub out DataViews

### PR DESCRIPTION
**Pull request by allilevine**
Related to https://github.com/Automattic/loop/issues/192

## Proposed Changes

- When the `reader/recent-feed-overhaul` flag is enabled, show a stubbed out DataViews that renders recent posts you can click on.

https://github.com/user-attachments/assets/49f6c643-8a48-42ac-981f-033d190f4da3

How this will look, for translation:
378424372-18a5ba89-dc7c-44c1-98f7-0524d8480955

## Why are these changes being made?

* pe7F0s-2iv-p2

## Testing Instructions

* Go to `/read?flags=reader/recent-feed-overhaul`
* Click site titles to load post titles.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/reader/following/main.jsx [hunk 1]
```diff
@@ -8,28 +9,32 @@ import ReaderOnboarding from 'calypso/reader/onboarding';
...
-			>
-				<BloganuaryHeader />
-				<NavigationHeader
-					title={ translate( 'Recent' ) }
-					subtitle={ translate( "Stay current with the blogs you've subscribed to." ) }
-					className={ clsx( 'following-stream-header', {
-						'reader-dual-column': props.width > WIDE_DISPLAY_CUTOFF,
-					} ) }
-				/>
+			{ config.isEnabled( 'reader/recent-feed-overhaul' ) ? (
```
Comment (allilevine) [inline client/reader/following/main.jsx:18]: The only change here is that I'm checking the feature flag and rendering the new component if enabled.

File: client/reader/recent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	const defaultLayouts = [
+		{
+			label: translate( 'List' ),
+			icon: 'list-view',
+		},
+	];
+
+	const fetchData = useCallback( () => {
+		dispatch( viewStream( 'following', window.location.pathname ) as AnyAction );
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (allilevine) [inline client/reader/recent/index.tsx:63]: We had to make an exception for this in Reader Onboarding too. Maybe we need to look at the typing on state management.

**Review by DustyReagan — APPROVED** [review]
Looks like a good start to me!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16986659

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @allilevine, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `All Recent`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16986659).

**Pull request merged**